pandas/_libs/lib.pyx

```diff
@@ -1110,7 +1110,7 @@ _TYPE_MAP = {
     "complex64": "complex",
     "complex128": "complex",
     "c": "complex",
-    "string": "string",
+    str: "string",
```

Comment:
```suggestion
    "string": "string",
    str: "string",
```

We could keep both, and that should address the small slowdown for inferring an array with `dtype="string"` (as it will first check the name, and only then the `dtype.type`). 

(but again, the infer_dtype is mostly used to infer actual lists or object dtype arrays, the inferring of an array with already a proper dtype is fast anyway, so I don't think this small difference matters much)

Reply:
changed to keep both for now, so as not to affect performance of object backed StringArray.